R/CDPSession.R

```diff
@@ -1,14 +1,47 @@
 #' @include EventEmitter.R
-NULL
-
-#' Create a websocket connexion using the Chrome DevTools Protocol
-#'
-#' @name CDPSession
+#' @include domain.R
+#' @include CDProtocol.R
+#' @include utils.R
 NULL
 
 #' @export
-CDPSession <- R6::R6Class(
-  "CDPSession",
+CDPSession <- function(
+  host = "localhost", port = 9222, secure = FALSE, ws_url = NULL,
+  autoConnect = FALSE, local = FALSE
+) {
+  url <- build_url(host, port, secure)
+  protocol <- CDProtocol$new(url = url, local = local)
+  if(is.null(ws_url)) {
+    ws_url <- chr_get_ws_addr(port = port)
+  }
+  CDPSession <- R6::R6Class(
+    "CDPSession",
+    inherit = CDPConnexion,
+    public = list(
+      initialize = function(ws_url, protocol, autoConnect) {
+        super$initialize(ws_url = ws_url, autoConnect = FALSE)
+        self$.__protocol__ <- protocol
+        lapply(protocol$domains, function(name) self[[name]] <- domain(self, name))
```

Comment:
it is more a question for me I guess but for this kind of stuff where we iterate over a list to return nothing like `purrr::walk` (here to assign) I would you a for loop. Assignment inside a `lapply` is not so common, and does not work in Global env. I am still 🤔 why it is working here.

Reply:
You're :100: right. I've hesitated to use `for` loops. I don't know why `lapply` is working here.
Do not hesitate to change to use of `for` loops.

Reply:
Ok I'll do that

Reply:
I will deal with that after merging in #41